client/config/toml.go

```diff
@@ -38,13 +52,66 @@ func writeConfigToFile(configFilePath string, config *ClientConfig) error {
 		return err
 	}
 
+	// Check if config file exists
+	if _, err := os.Stat(configFilePath); os.IsNotExist(err) {
+		fmt.Println("creating new node configuration in", filepath.Dir(filepath.Dir(configFilePath)))
+	}
```

Comment:
You can hit other permissions other than not exist, could be a permission access or a linking error. Perhaps
```go
switch _, err := os.Stat(configFilePath);  {
case os.IsNotExist(err):
     fmt.Println("creating new node configuration in", filepath.Dir(filepath.Dir(configFilePath)))
case err != nil:
     return fmt.Errorf("unexpected error while stating config path: %w", err)
}
```